docs/new-components.md

```diff
@@ -0,0 +1,205 @@
+component, you may [specify a local folder using the `replaces` option of the
+builder](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder#configuration)
+to include it in your build.
+
+If you want to **publish** your component for other people to try it out, you can do so for free
+using Github or other git forges. To do so, you need to [publish your component as a Go
+module](https://go.dev/doc/modules/publishing). You can publish multiple components from a single
+repository by including the path to the component in the tag: for example, the [`filelog`
+receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver#file-log-receiver)
+v0.139.0 version is [available as a Go
+module](https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filelogreceiver@v0.139.0)
+because we pushed the
+[`receiver/filelogreceiver/v0.139.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver)
+git tag to this repository. We recommend you make a release at least every time there is a breaking
+change on any of the Go modules you depend on.
+
+Finally, to **distribute** your component and make sure others can easily discover it, [add it to
+the OpenTelemetry registry](https://opentelemetry.io/ecosystem/registry/adding/).
+
+If you think your component fits the requisites to be on the opentelemetry-collector-contrib
+repository, you may choose to donate it following the steps on the next section. If you have
+published your component outside of this repository, your activity, community and popularity within
+your own repository will help you make the case for the component to be accepted.
+
+## Adding your component to the opentelemetry-collector-contrib repository
+
+You may donate an existing component or propose a whole new one. We recommend that you **start by
+building and hosting your component outside of the opentelemetry-collector-contrib repository**.
+After you have gotten some real usage you can contribute your component to this repository.
+
+When you are ready to propose adding your component to this repository, [open an
+issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/new?assignees=&labels=Sponsor+Needed%2Cneeds+triage&projects=&template=new_component.yaml&title=New+component%3A+)
+providing the following information:
+
+* Who's the **sponsor** for your component. A sponsor is an approver or maintainer who will be the
+  official reviewer of the code and a code owner for the component. You will need to find a sponsor
+  for the component in order for it to be accepted. Ideally, the sponsor is from a different company
+  than you. If the person proposing the component and the sponsor are
+  from the same company, ensure that you have approval on the overall proposal from other approvers
+  and maintainers.
```

Comment:
It would be nice to include some information on how to find a sponsor if you're new to the community.